src/sentry/api/endpoints/project_rule_enable.py

```diff
@@ -0,0 +1,61 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log
+from sentry.api.api_owners import ApiOwner
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.endpoints.project_rules import find_duplicate_rule
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.constants import ObjectStatus
+from sentry.models import Rule
+
+
+@region_silo_endpoint
+class ProjectRuleEnableEndpoint(ProjectEndpoint):
+    owner = ApiOwner.ISSUES
+    permission_classes = (ProjectAlertRulePermission,)
+
+    def put(self, request: Request, project, rule_id) -> Response:
+        try:
+            rule = Rule.objects.get(id=rule_id)
```

Comment:
seems like as written this allows for a direct object reference. need to validate the project_id on the query. https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html

Reply:
heh I was just thinking that as I was re-reading it 😅